dash/frontend/src/app/modules/private/pages/image/image-scan-result/image-scan-result.component.ts

```diff
@@ -139,15 +139,20 @@ export class ImageScanResultComponent implements OnInit, AfterViewInit, OnDestro
           } else {
             this.displayImageScanSpinner$ = of(false);
           }
-          return this.imageService.getNamespaceByImageHash(this.dataSource.dockerImageId);
+          if (!this.dataSource.dockerImageId) {
+            return [];
+          }
+          return this.imageService.getNamespaceByImageHash(this.dataSource.dockerImageId).pipe(take(1));
```

Comment:
Since you are switchMapping into this observable, you don't have to close it with a take, as the take in the outer pipe will close it. (It also doesn't hurt anything though)